src/components/Main/Members/Members.tsx

```diff
@@ -111,83 +77,55 @@ export const Members = () => {
     email: string,
     password: string,
   ) => {
-    setIsAddUserDialogOpen(false);
     if (shouldAdd && firstName && lastName && email && password && isValidEmail(email)) {
       dispatch(onAddMember(firstName, lastName, email, password));
     }
+    setIsAddMemberDialogOpen(false);
   };
 
-  const handleRowClick = (event: React.MouseEvent<unknown>, row: MemberStateData) => {
+  const handleRowClick = (
+    event: React.MouseEvent<unknown>,
+    { userId: id }: MemberStateData,
+  ) => {
     event.stopPropagation();
-    const selectedIndex = selectedRows.indexOf(row.userId);
-    let newSelectedRows: number[] = [];
-    if (event.ctrlKey) {
-      if (selectedIndex === -1) {
-        newSelectedRows = [...selectedRows, row.userId];
-      } else if (selectedIndex === 0) {
-        newSelectedRows = selectedRows.slice(1); // all but first row
-      } else if (selectedIndex === selectedRows.length - 1) {
-        newSelectedRows = selectedRows.slice(0, -1); // all but last row
-      } else if (selectedIndex > 0) {
-        newSelectedRows = [
-          ...selectedRows.slice(0, selectedIndex),
-          ...selectedRows.slice(selectedIndex + 1),
-        ];
-      }
-    } else {
-      if (selectedIndex === -1) newSelectedRows = [row.userId];
-      else
-        newSelectedRows = [
-          ...selectedRows.slice(0, selectedIndex),
-          ...selectedRows.slice(selectedIndex + 1),
-        ];
-    }
-    // dispatch(onLoadUser(row));
-    setSelectedUser(row);
-    setSelectedRows(newSelectedRows);
+    setLastSelected(id);
+    const updatedRows: number[] = event.shiftKey
+      ? makeNewRows(lastSelected, id, data)
+      : [id];
+    selectedRows.includes(id)
+      ? setSelectedRows(selectedRows.filter((rowId) => !updatedRows.includes(rowId)))
+      : setSelectedRows([...selectedRows, ...updatedRows]);
```

Comment:
essentially this is a select/deselect function. when the row is clicked, if the row already exists in the selectedRows, it will remove, otherwise it will add rows. the updatedRows is determined depending if the shift key is pressed, just like when you're on your email and you're trying to select or deselect many at a time.

that being said, this is still incomplete as it doesn't consider rows already selected/deselected in between two index numbers. but I plan on doing this at a later time